dialect/pgdialect/alter_table.go

```diff
@@ -0,0 +1,248 @@
+	fmter := m.db.Formatter()
+
+	// Append ALTER TABLE statement to the enclosed query bytes []byte.
+	appendAlterTable := func(query []byte, fqn schema.FQN) []byte {
+		query = append(query, "ALTER TABLE "...)
+		query, _ = fqn.AppendQuery(fmter, query)
+		return append(query, " "...)
+	}
+
+	switch change := operation.(type) {
+	case *migrate.CreateTableOp:
+		return m.AppendCreateTable(b, change.Model)
+	case *migrate.DropTableOp:
+		return m.AppendDropTable(b, change.FQN)
+	case *migrate.RenameTableOp:
+		b, err = m.renameTable(fmter, appendAlterTable(b, change.FQN), change)
+	case *migrate.RenameColumnOp:
+		b, err = m.renameColumn(fmter, appendAlterTable(b, change.FQN), change)
+	case *migrate.AddColumnOp:
+		b, err = m.addColumn(fmter, appendAlterTable(b, change.FQN), change)
+	case *migrate.DropColumnOp:
+		b, err = m.dropColumn(fmter, appendAlterTable(b, change.FQN), change)
+	case *migrate.AddPrimaryKeyOp:
+		b, err = m.addPrimaryKey(fmter, appendAlterTable(b, change.FQN), change.PrimaryKey)
+	case *migrate.ChangePrimaryKeyOp:
+		b, err = m.changePrimaryKey(fmter, appendAlterTable(b, change.FQN), change)
+	case *migrate.DropPrimaryKeyOp:
+		b, err = m.dropConstraint(fmter, appendAlterTable(b, change.FQN), change.PrimaryKey.Name)
+	case *migrate.AddUniqueConstraintOp:
+		b, err = m.addUnique(fmter, appendAlterTable(b, change.FQN), change)
+	case *migrate.DropUniqueConstraintOp:
+		b, err = m.dropConstraint(fmter, appendAlterTable(b, change.FQN), change.Unique.Name)
+	case *migrate.ChangeColumnTypeOp:
+		b, err = m.changeColumnType(fmter, appendAlterTable(b, change.FQN), change)
+	case *migrate.AddForeignKeyOp:
+		b, err = m.addForeignKey(fmter, appendAlterTable(b, change.FQN()), change)
+	case *migrate.DropForeignKeyOp:
+		b, err = m.dropConstraint(fmter, appendAlterTable(b, change.FQN()), change.ConstraintName)
+	default:
+		return nil, fmt.Errorf("append sql: unknown operation %T", change)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("append sql: %w", err)
+	}
+	return b, nil
+}
+
+func (m *migrator) renameTable(fmter schema.Formatter, b []byte, rename *migrate.RenameTableOp) (_ []byte, err error) {
+	b = append(b, "RENAME TO "...)
+	b = fmter.AppendName(b, rename.NewName)
+	return b, nil
+}
+
+func (m *migrator) renameColumn(fmter schema.Formatter, b []byte, rename *migrate.RenameColumnOp) (_ []byte, err error) {
+	b = append(b, "RENAME COLUMN "...)
+	b = fmter.AppendName(b, rename.OldName)
+
+	b = append(b, " TO "...)
+	b = fmter.AppendName(b, rename.NewName)
+
+	return b, nil
+}
+
+func (m *migrator) addColumn(fmter schema.Formatter, b []byte, add *migrate.AddColumnOp) (_ []byte, err error) {
+	b = append(b, "ADD COLUMN "...)
+	b = fmter.AppendName(b, add.Column)
+	b = append(b, " "...)
+
+	colDef, ok := add.ColDef.(sqlschema.ColumnDefinition)
+	if !ok {
+		return nil, fmt.Errorf("column %q does not implement sqlschema.ColumnDefinition, got %T", add.ColDef.GetName(), add.ColDef)
```

Comment:
ColumnDefinition is a struct so it is actually ColumnDefinition which implements Column. I will change these names...

Reply:
Thank you